### PR TITLE
SOF-2249 Reconnect to Panel on PanelConfiguration updates

### PR DESCRIPTION
**Pull request by LindvedKrvang**
When a PanelConfiguration is updated, we disconnect the current Panel for it, delete the Panel and create a new Panel from the updated PanelConfiguration.

The reason for disconnecting and discarding is, that the new PanelConfiguration could be incompatible with the current Panel (maybe it changed from Skaarhoj to StreamDeck - not currently supported, but it showcases the usecase).

### CODE REVIEW

File: src/business-logic/services/panel-manager-implementation.ts [hunk 1]
```diff
@@ -34,6 +64,10 @@ export class PanelManagerImplementation implements PanelManager {
         this.disconnectFromPanel(panelEvent.panelConfigurationId)
         return
       }
+      if (this.isPanelConfigurationUpdatedEvent(panelEvent)) {
+        this.reconnectToPanel(panelEvent.panelConfiguration)
+        return
```
Comment (MadsThy) [inline src/business-logic/services/panel-manager-implementation.ts:69]: im pretty sure you don't need this return 😄 
```suggestion
```

**Review by MadsThy — APPROVED** [review]

File: src/business-logic/services/panel-manager-implementation.ts (hunk 1 above)
Comment (LindvedKrvang) [reply src/business-logic/services/panel-manager-implementation.ts:69]: But now the return from the delete was necessary ;)

Comment (LindvedKrvang) [reply src/business-logic/services/panel-manager-implementation.ts:69]: This is going to get an overhaul in SOF-2261.
In my current implementation I'm breaking the rules/barriers of our architecture which is not allowed... :p

File: src/business-logic/services/panel-manager-implementation.ts [hunk 2]
```diff
@@ -24,7 +25,36 @@ export class PanelManagerImplementation implements PanelManager {
...
+    try {
+      await this.connectToPanels()
+    } catch (error) {
+      this.logger.data(error).error('Error while connecting to Panels')
+    }
+  }
+
+  private async connectToPanels(): Promise<void> {
+    const panelConfigurations: PanelConfiguration[] = await this.panelConfigurationRepository.getPanelConfigurations()
+    panelConfigurations.map(this.connectToPanel.bind(this))
```
Comment (KvelaGorrrrnio) [inline src/business-logic/services/panel-manager-implementation.ts]: `forEach` can be used here, since nothing is returned.

**Review by KvelaGorrrrnio — APPROVED** [review]

File: src/business-logic/services/panel-manager-implementation.ts (hunk 2 above)
Comment (LindvedKrvang) [reply src/business-logic/services/panel-manager-implementation.ts]: Updated

**Pull request merged**